pkg/hints/usort_hints.go

```diff
@@ -0,0 +1,239 @@
+package hints
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/lambdaclass/cairo-vm.go/pkg/lambdaworks"
+	"github.com/lambdaclass/cairo-vm.go/pkg/types"
+	. "github.com/lambdaclass/cairo-vm.go/pkg/vm"
+	"github.com/lambdaclass/cairo-vm.go/pkg/vm/memory"
+
+	. "github.com/lambdaclass/cairo-vm.go/pkg/hints/hint_utils"
+	"github.com/pkg/errors"
+)
+
+// SortFelt implements sort.Interface for []lambdaworks.Felt
+type SortFelt []lambdaworks.Felt
+
+func (s SortFelt) Len() int      { return len(s) }
+func (s SortFelt) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+func (s SortFelt) Less(i, j int) bool {
+	a, b := s[i], s[j]
+
+	return a.Cmp(b) == -1
+}
+
+// Implements hint:
+// %{ vm_enter_scope(dict(__usort_max_size = globals().get('__usort_max_size'))) %}
+func usort_enter_scope(executionScopes *types.ExecutionScopes) error {
+	usort_max_size_interface, err := executionScopes.Get("usort_max_size")
+
+	if err != nil {
+		executionScopes.EnterScope(make(map[string]interface{}))
+		return nil
+	}
+
+	usort_max_size, cast_ok := usort_max_size_interface.(uint64)
+
+	if !cast_ok {
+		return errors.New("Error casting usort_max_size into a uint64")
+	}
+
+	scope := make(map[string]interface{})
+	scope["usort_max_size"] = usort_max_size
+	executionScopes.EnterScope(scope)
+
+	return nil
+}
+
+func usort_body(ids IdsManager, executionScopes *types.ExecutionScopes, vm *VirtualMachine) error {
+
+	input_ptr, err := ids.GetRelocatable("input", vm)
+	if err != nil {
+		return err
+	}
+
+	input_len, err := ids.GetFelt("input_len", vm)
+
+	if err != nil {
+		return err
+	}
+	input_len_u64, err := input_len.ToU64()
+
+	if err != nil {
+		return err
+	}
+
+	usort_max_size, err := executionScopes.Get("usort_max_size")
+
+	if err == nil {
+		usort_max_size_u64, cast_ok := usort_max_size.(uint64)
+
+		if !cast_ok {
+			return errors.New("Error casting usort_max_size into a uint64")
+		}
+
+		if input_len_u64 > usort_max_size_u64 {
+			return errors.New(fmt.Sprintf("usort() can only be used with input_len<= %v. Got: input_len=%v.", usort_max_size_u64, input_len_u64))
+		}
+	}
+
+	positions_dict := make(map[lambdaworks.Felt][]uint64)
+
+	for i := uint64(0); i < input_len_u64; i++ {
+
+		val, err := vm.Segments.Memory.GetFelt(input_ptr.AddUint(uint(i)))
+
+		if err != nil {
+			return err
+		}
+
+		positions_dict[val] = append(positions_dict[val], i)
+	}
+	executionScopes.AssignOrUpdateVariable("positions_dict", positions_dict)
+
+	output := make([]lambdaworks.Felt, 0, len(positions_dict))
+
+	for key := range positions_dict {
+		output = append(output, key)
+	}
+
+	sort.Sort(SortFelt(output))
+
+	output_len := memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(uint64((len(output)))))
+	ids.Insert("output_len", output_len, vm)
+
+	output_base := vm.Segments.AddSegment()
+
+	for i := range output {
+		vm.Segments.Memory.Insert(output_base.AddUint(uint(i)), memory.NewMaybeRelocatableFelt(output[i]))
```

Comment:
Please handle the error returned by Memory.Insert

Reply:
Done
[fc6134f](https://github.com/lambdaclass/cairo-vm_in_go/pull/244/commits/fc6134f314b97ed3435de42c78e75cf28919082a)